src/engine/specification.md

```diff
@@ -205,17 +207,19 @@ This structure contains the attributes required to initiate a payload build proc
 
 1. The values `(forkchoiceState.headBlockHash, forkchoiceState.finalizedBlockHash)` of this method call map on the `POS_FORKCHOICE_UPDATED` event of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and **MUST** be processed according to the specification defined in the EIP.
 
-2. All updates to the forkchoice state resulting from this call **MUST** be made atomically.
+1. All updates to the forkchoice state resulting from this call **MUST** be made atomically.
+
+1. Client software **MUST** return `{status: SUCCESS, payloadId: null}` if `payloadAttributes` is `null`.
```

Comment:
We should clarify that this isn't returned *no matter what* if `payloadAttribues` is `null` but only if null and successful

```suggestion
1. Client software **MUST** return `{status: SUCCESS, payloadId: null}` if `payloadAttributes` is `null` and fork choice update is successful.
```

Reply:
Actually, maybe we should reorder this statement with the next one.

* `{SYNCING, null}` if not enough info to process
* `{SUCCESS, null}` if no payload and success
* `{SUCCESS, id}` if payload and success

Reply:
That way it acts a blocking condition on the next two success conditions